src/sql/base/browser/ui/selectBox/selectBox.ts

```diff
@@ -66,12 +66,12 @@ export class SelectBox extends vsSelectBox {
 
 	constructor(options: SelectOptionItemSQL[] | string[], selectedOption: string, contextViewProvider: IContextViewProvider, container?: HTMLElement, selectBoxOptions?: ISelectBoxOptions) {
 		let optionItems: SelectOptionItemSQL[];
-		if (Array.isArray<string>(options)) {
-			optionItems = (options as string[]).map(o => {
+		if (typeof (options[0]) === 'string' && Array.isArray<string>(options)) {
```

Comment:
apparently the Array.isArray<string>(options) did not work as expected, we probably don't need it. also we need to check the array length.

Reply:
It does do type coercion, so it's still useful.